src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -870,10 +870,19 @@ namespace winrt::TerminalApp::implementation
                     {
                         focusedObject = winrt::Windows::UI::Xaml::Media::VisualTreeHelper::GetParent(focusedElement);
 
-                        // We were unable to find a focused object. Default to the xaml root so that the alt+space menu still works.
+                        // We were unable to find a focused object. Give the
+                        // TerminalPage one last chance to let the alt+space
+                        // menu still work.
+                        //
+                        // We return always, because the TerminalPage handler
+                        // will return false for just a bare `alt` press, and
+                        // don't want to go around the loop again.
                         if (!focusedObject)
                         {
-                            focusedObject = _root.try_as<IInspectable>();
+                            if (auto keyListener{ _root.try_as<IDirectKeyListener>() })
+                            {
+                                return keyListener.OnDirectKeyEvent(vkey, scanCode, down);
+                            }
                         }

```

Comment:
I'm not sure if it would make more sense, but technically you can put this fallback outside the loop, at the end of this function. I guess it would make some sense, given that it only accesses `_root` and has nothing to do with the actual UI tree anymore.